hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoCreateHandler.java

```diff
@@ -92,25 +112,23 @@ public void handle(@NonNull final HandleContext handleContext) {
         final var txnBody = handleContext.body();
         final var op = txnBody.cryptoCreateAccount();
 
-        // validate fields in the transaction body that involves checking with
-        // dynamic properties or state
-        final ResponseCodeEnum validationResult = validateSemantics();
-        if (validationResult != OK) {
-            throw new HandleException(validationResult);
-        }
+        final var accountStore = handleContext.writableStore(WritableAccountStore.class);
 
         // FUTURE: Use the config and check if accounts can be created.
         //  Currently, this check is being done in `finishCryptoCreate` before `commit`
 
         // validate payer account exists and has enough balance
-        final var accountStore = handleContext.writableStore(WritableAccountStore.class);
         final var payer = accountStore.getForModify(
                 txnBody.transactionIDOrElse(TransactionID.DEFAULT).accountIDOrElse(AccountID.DEFAULT));
 
         validateTrue(payer != null, INVALID_PAYER_ACCOUNT_ID);
         final long newPayerBalance = payer.tinybarBalance() - op.initialBalance();
         validatePayer(payer, newPayerBalance);
```

Comment:
Is there a reason these checks aren't inside `validateSemantics(...)`?

Reply:
Because payer is used later in the code for modifiedPayer and be put to accountStore.

Reply:
It would be nice to group these validations with the other validations in `validateSemantics(...)` so they're all in the same place, and then return whatever is needed wrapped in a private object. 

For example:
```java
private ValidationResult validateSemantics(@NonNull final ...) {
    var payer = accountStore.get(accountId);
    validateTrue(payer != null, INVALID_ACCOUNT_ID);

    var token = tokenStore.get(tokenId);
    validateTrue(token != null, INVALID_TOKEN_ID);

    // other validations, etc. ... 

    // Now, at the end:
    return new ValidationResult(payer, token, ...);
}

private record ValidationResult(Account payer, Token token, ...) {}
```

Reply:
I updated the code and put all the validations in validateSemantics(). Since WritableAccountStore is passed in, I can update accountStore there as well. No need to return ValidationResult.